worlds/sc2/__init__.py

```diff
@@ -196,6 +196,23 @@ def fill_slot_data(self) -> Mapping[str, Any]:
         if SC2Campaign.HOTS not in enabled_campaigns:
             slot_data["kerrigan_presence"] = KerriganPresence.option_not_present
 
+        if get_option_value(self, "mission_order_scouting") > 0:
+            mission_item_classification: Dict[str, ItemClassification] = {}
+            for location in self.multiworld.get_locations(self.player):
+                # Victory location are duplicated as event with the Beat prefix. Ignore them.
+                if not location.name.startswith("Beat"):
+                    loc: Location = self.multiworld.get_location(location.name, self.player)
+                    # Ensure that if there are multiple items given for finishing a mission and that at least 
+                    # one is progressive, the flag kept is progressive.
+                    # Assuming that the location added are always after the basic one and named Cache. 
+                    if " Cache (" in location.name:
+                        location_name = location.name.split(" Cache")[0]
+                        if mission_item_classification[location_name] != 1:
+                            mission_item_classification[location_name] = loc.item.classification.as_flag() 
+                    else:
+                        mission_item_classification[location.name] = loc.item.classification.as_flag()
```

Comment:
String comparison was removed and explicit integer comparison was replaced with enum approach as suggested elsewhere. Also, the `loc: Location = self.multiworld.get_location(location.name, self.player)` returned the same thing as `location` so it was removed.

Notes on removal of string comparison:
The "Beat" one was replaced with `is_event`, which was the intent of this `if`. Checking "Best" location was "acceptable" since they are the only event in SC2 world, but `is_event` is more robust.

Detection of victory cache via string was replaced by using `lookup_location_id_to_type` which was already created in `locations.py` and the assumption that all Victory cache locations are preceded by the basic Victory location. This last assumption seems to be valid as long as locations are sorted by their ID and that `VICTORY_CACHE_OFFSET` is positive, which seems acceptable.

This last modification was an attempt to limit the creation of a new lookup table for that. If the assumptions are problematic, I will try something else.

Reply:
The changed code looks better (more readable and immune to location naming scheme changes)